packages/json-deserializer/src/JsonDeserializer.test.ts

```diff
@@ -0,0 +1,98 @@
+import { jsonDeserializer } from "./JsonDeserializer";
+import { createContext, createEvent } from "@lambda-middleware/utils";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { APIGatewayProxyObjectEvent } from "./types/APIGatewayProxyObjectEvent";
+
+describe("jsonDeserializer", () => {
+  let processedEvent: APIGatewayProxyObjectEvent<APIGatewayProxyEvent>;
+
+  const dummyHandler = async (
+    event: APIGatewayProxyObjectEvent<APIGatewayProxyEvent>
+  ) => {
+    processedEvent = event;
+    return {} as APIGatewayProxyResult;
+  };
+  const handlerWithMiddleware = jsonDeserializer()(dummyHandler);
+
+  const testObject = {
+    a: "json",
+    object: {
+      with: ["multiple", "types"],
+      ofProperties: {
+        numbers: 2345,
+        booleans: true,
+        stringValues: "testString",
+        nulls: null,
+        undefineds: undefined,
+        andArrays: ["ofStrings", 42, null],
+      },
+    },
+  };
+
+  it("should deserialize objects if the event has a json content-type header", async () => {
+    const event: APIGatewayProxyEvent = createEvent({
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(testObject),
+    });
+
+    await handlerWithMiddleware(event, createContext());
+    expect(processedEvent.body).toMatchObject(
+      JSON.parse(JSON.stringify(testObject))
+    );
+  });
+
+  describe("should set the request body to null if:", () => {
+    it.each([
+      ["null headers", null],
+      ["null content type", { "Content-Type": null }],
+      ["a non-json content-type header", { "Content-Type": "text/plain" }],
+    ])("the event has %s", async (_: string, headers: any) => {
+      const event = createEvent({
+        headers,
+        body: JSON.stringify(testObject),
+      });
+
+      await handlerWithMiddleware(event, createContext());
+      expect(processedEvent.body).toBeNull();
+    });
+  });
+
+  it("should decode the request body before deserializing if isBase64Encoded is true", async () => {
+    const event = createEvent({
+      headers: { "Content-Type": "application/json" },
+      body: Buffer.from(JSON.stringify(testObject)).toString("base64"),
+      isBase64Encoded: true,
+    });
+
+    await handlerWithMiddleware(event, createContext());
+    expect(processedEvent.body).toMatchObject(
+      JSON.parse(JSON.stringify(testObject))
+    );
+  });
+
+  it("should leave the request body as null if body is null", async () => {
+    const event = createEvent({
+      headers: { "Content-Type": "application/json" },
+      body: null,
+    });
+
+    await handlerWithMiddleware(event, createContext());
+    expect(processedEvent.body).toBeNull();
+  });
+
+  it("should throw a RequestBodyNotJsonError if the request body cannot deserialize", async () => {
```

Comment:
I think at the time I was thinking "what about GET scenarios" if I were to setup my middlewares generically outside of the handlers and just pass the handler to a factory function that wraps a handler in standard middlewares for the project if you'd like to standardise them rather than repeating adding all the layers in handlers where it's needed.  There's no harm as such in returning null in this case.

Thinking about this, I think maybe adding a separate `bodyObject` rather than replacing `body` might be better with this behaviour as the null functionality is halfway towards making it more "compatible" with varying usage.

Reply:
I've not changed the behaviour for the reasons above, but I've implemented the `bodyObject` instead of overwriting the body, that will probably make the behaviour make more sense.